scripts/gen-upgrade-proposal.sh

```diff
@@ -0,0 +1,43 @@
+#! /bin/bash
+set -ueo pipefail
+
+cat <<EOF 1>&2
+------------------------------------------------------------
+This script shows a command for a software upgrade proposal which is
+compatible with the Cosmovisor found at:
+https://github.com/agoric-labs/cosmos-sdk/tree/Agoric/cosmovisor#readme
+------------------------------------------------------------
+EOF
+
+COMMIT_ID=$(git rev-parse HEAD)
+ZIPURL=https://github.com/Agoric/agoric-sdk/archive/$COMMIT_ID.zip
+
+echo "Verifying archive is at $ZIPURL..." 1>&2
+zipfile=$(mktemp)
+trap 'rm -f "$zipfile"' EXIT
+curl -L "$ZIPURL" -o "$zipfile"
+
+echo "Generating SHA-256 checksum..." 1>&2
+checksum=sha256:$(shasum -a 256 "$zipfile" | cut -d' ' -f1)
+
+info="{\"binaries\":{\"any\":\"$ZIPURL?checksum=$checksum\"}}"
+
+cat <<EOF 1>&2
+------------------------------------------------------------
+Here is the skeleton of the recomended upgrade proposal command.
+You'll need to fill in the details:
+  <UPGRADE-NAME> is the name of the upgrade built into the agd binary
```

Comment:
Note that this will be provided by the organization that authored the upgrade.

Reply:
I decided to remove these specific details in favour of more generic guidance, to reinforce that software upgrades are a power tool.